includes/class-wc-payments-order-service.php

```diff
@@ -438,7 +422,7 @@ function ( array $event ) {
 	 * @return string
 	 */
 	public function get_intent_id_for_order( $order ) : string {
-		$order = wc_get_order( $order );
```

Comment:
`wc_get_order` if the order is not found for some reason, so a new method was created to handle getting the order. 

Discussion about this here: p1679590206525959-slack-C01B8KNUYSW